File: SWIG/Examples/test-suite/lua/primitive_ref_runme.lua
require("import")	-- the import fn
import("primitive_ref",true)	-- import code into global namespace

assert(ref_int(3)==3)

assert(ref_uint(3) == 3)

assert(ref_short(3) == 3)

assert(ref_ushort(3) == 3)

assert(ref_long(3) == 3)

assert(ref_ulong(3) == 3)

assert(ref_schar(3) == 3)

assert(ref_uchar(3) == 3)

assert(ref_float(3.5) == 3.5)

assert(ref_double(3.5) == 3.5)

assert(ref_bool(true) == true)

assert(ref_char('x') == 'x')

assert(ref_over(0) == 0)

a=A(12)
assert(ref_over(a)==12)

File: Examples/test-suite/perl5/primitive_ref_runme.pl
use primitive_ref;

if (primitive_ref::ref_int(3) != 3) {
    die "ref_int failed!\n";
}
if (primitive_ref::ref_uint(3) != 3) {
    die "ref_uint failed!\n";
}
if (primitive_ref::ref_short(3) != 3) {
    die "ref_short failed!\n";
}
if (primitive_ref::ref_ushort(3) != 3) {
    die "ref_ushort failed!\n";
}
if (primitive_ref::ref_long(3) != 3) {
    die "ref_long failed!\n";
}
if (primitive_ref::ref_ulong(3) != 3) {
    die "ref_ulong failed!\n";
}
if (primitive_ref::ref_schar(3) != 3) {
    die "ref_schar failed!\n";
}
if (primitive_ref::ref_uchar(3) != 3) {
    die "ref_uchar failed!\n";
}
if (primitive_ref::ref_bool(1) != 1) {
    die "ref_bool failed!\n";
}
if (primitive_ref::ref_float(3.5) != 3.5) {
    die "ref_float failed!\n";
}
if (primitive_ref::ref_double(3.5) != 3.5) {
    die "ref_double failed!\n";
}
if (primitive_ref::ref_char('x') != 'x') {
    die "ref_char failed!\n";
}
